.github/workflows/changeset.yml

```diff
@@ -0,0 +1,72 @@
+name: Create Change Set
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    branches:
+      - main
+
+jobs:
+  create-change-sets:
+    name: Create Change Sets
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        environment: [Beta, Prod]
+    environment: ${{ matrix.environment }}
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js and AWS CDK
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18.x
+
+      - name: Install dependencies and build project
+        run: |
+          npm install
+          npm run build
+
+      - name: Install AWS CDK
+        run: npm install -g aws-cdk@2.149.0
+
+      - name: Assume IAM Role for Beta
+        uses: aws-actions/configure-aws-credentials@v4.0.2
+        with:
+          role-to-assume: ${{ secrets.CHANGESET_ROLE }}
+          aws-region: us-east-1
+
+      - name: Create Change Set for Beta
+        id: cdk_diff
+        run: |
+          cdk acknowledge 30717
+          echo "diff_output<<EOF" >> $GITHUB_OUTPUT
+          echo "## CI-Config Stack Changeset" >> $GITHUB_OUTPUT
+          npm run cdk diff -- OpenSearch-CI-Config-${{ matrix.environment }} -c useSsl=true -c serverAccessType=prefixList -c restrictServerAccessTo=${{ secrets.PREFIX_LIST }} | sed -E 's/[0-9]{12}/[MASKED]/g' >> $GITHUB_OUTPUT
+          
+          echo "" >> $GITHUB_OUTPUT
+          echo "## CI Stack ChangeSet ${{ matrix.environment }}" >> $GITHUB_OUTPUT
+          npm run cdk diff -- OpenSearch-CI-${{ matrix.environment }} -c useSsl=true -c authType=github -c dataRetention=true -c macAgent=true -c useProdAgents=true -c enableViews=true -c ignoreResourcesFailures=false -c serverAccessType=prefixList -c restrictServerAccessTo=${{secrets.PREFIX_LIST}} | sed -E 's/[0-9]{12}/[MASKED]/g' >> $GITHUB_OUTPUT
```

Comment:
Lets not use context variables. Maybe create a new PR with properties set as desired?
Lets get that PR in first which will simply the deployment and diff commands significantly.